homeassistant/config.py

```diff
@@ -1247,9 +1225,16 @@ def _get_log_message_and_stack_print_pref(
             show_stack_trace = True
         return (log_message, show_stack_trace, placeholders)
 
-    assert isinstance(log_message_show_stack_trace, tuple)
+    # Get the log message from the translation cache
+    log_message = str(
+        HomeAssistantError(
+            translation_domain="homeassistant",
+            translation_key=platform_exception.translation_key,
+            translation_placeholders=placeholders,
+        )
```

Comment:
It's seems unnecessary to create an object here, only to immediately discard it. Can't we just call `async_get_exception_message`?